upb/def.c

```diff
@@ -2245,6 +2241,10 @@ static void create_fielddef(
     upb_value v, field_v, json_v, existing_v;
     size_t json_size;
 
+    if (field_number == 0 || field_number > UPB_MAX_FIELDNUMBER) {
```

Comment:
I vaguelly prefer <= so I don't need to remember if this is signed or not

Reply:
Done (actually I changed it to `int32_t` since the field is int32 and we should in fact check for and reject negative values here.